src/ArcgisAccount/accountManagerUtils.js

```diff
@@ -0,0 +1,236 @@
+import { UserSession } from '@esri/arcgis-rest-auth';
+import { getSelf } from '@esri/arcgis-rest-portal';
+import { request } from '@esri/arcgis-rest-request';
+
+import {
+  addAccountStorage,
+  getAccountManagerStorage
+} from './accountManagerStorageUtils';
+
+//** RestJS */
+/** Begin OAuth  */
+export const beginOAuthSignIn = async ({
+  clientId,
+  redirectUri,
+  portalUrl,
+  popup,
+  name,
+  setAccountManagerState
+}) => {
+  if (popup) {
+    try {
+      const portal = portalUrl
+        ? portalUrl
+        : 'https://www.arcgis.com/sharing/rest';
+      UserSession.beginOAuth2({
+        // register an app of your own to create a unique clientId
+        clientId: clientId,
+        redirectUri: redirectUri,
+        portal: portal,
+        popup: popup
```

Comment:
We could use Object Initializer shorthand here:
```suggestion
      UserSession.beginOAuth2({
        // register an app of your own to create a unique clientId
        clientId,
        redirectUri,
        portal,
        popup
```